test/EndToEndTests/Tests/Client/Build.Desktop/DeltaTests/DeltaTests.cs

```diff
@@ -22,217 +25,918 @@ public DeltaTests()
 
         }
 
-        private readonly string[] deltaMimeTypes = new string[]
-            {
-                MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata,
-                MimeTypes.ApplicationJson + MimeTypes.ODataParameterMinimalMetadata,
-                MimeTypes.ApplicationJson + MimeTypes.ODataParameterNoMetadata,
-            };
-
-        [Fact]
-        public void RequestDeltaLink()
+        public static readonly IEnumerable<object[]> DeltaMimeTypes = new List<object[]>
+        {
+            new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata },
+            new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterMinimalMetadata },
+            new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterNoMetadata },
+        };
+
+        [Theory]
+        [MemberData(nameof(DeltaMimeTypes))]
+        public void RequestDeltaLink(string mimeType)
         {
             var customersSet = Model.FindDeclaredEntitySet("Customers");
-            var customerType = customersSet.EntityType();
+            var customerType = customersSet.EntityType;
 
-            ODataMessageReaderSettings readerSettings = new ODataMessageReaderSettings() { BaseUri = ServiceBaseUri };
+            var readerSettings = new ODataMessageReaderSettings { BaseUri = ServiceBaseUri };
 
-            foreach (var mimeType in deltaMimeTypes)
-            {
-                var requestMessage = new HttpWebRequestMessage(new Uri(ServiceBaseUri.AbsoluteUri + "$delta", UriKind.Absolute));
-                requestMessage.SetHeader("Accept", mimeType);
-                var responseMessage = requestMessage.GetResponse();
-                Assert.Equal(200, responseMessage.StatusCode);
+            var requestMessage = new HttpWebRequestMessage(new Uri(ServiceBaseUri.AbsoluteUri + "$delta", UriKind.Absolute));

```

Comment:
Is this class from OData Client.. I thought we moved away from it?

Reply:
@ElizabethOkerio This class is not from the OData client library. It is defined in a test project for testing purposes https://github.com/OData/odata.net/blob/fce8fd9bf4942de679918ba3258d4b8dba7660f0/test/EndToEndTests/Tests/Client/Build.Desktop/Common/HttpWebRequestMessage.cs#L18. It implements `IODataRequestMessageAsync` and `IContainerProvider` interfaces.